src/Chainweb/Pact/TransactionExec.hs

```diff
@@ -339,7 +343,69 @@ applyLocal logger dbEnv gasModel pd spv cmdIn mc =
         Exec !pm -> return pm
         _ -> throwCmdEx "local continuations not supported"
 
-      checkTooBigTx gas0 gasLimit (applyPayload em) return
+      -- if network id is defined, assume public meta exists
+      -- otherwise, we just run the payload as is
+      case nid of
+        Nothing ->
+          locally txGasLimit (const 25000) $ applyPayload em
+        Just nid' -> do
+          now <- liftIO $ getCurrentTimeIntegral
+          validateLocalCmd v pscid now gas0 tenv cmd nid'
+
+-- | Validate public meta for commands passed to /local endpoint
+-- with defined network id. If id is undefined, these tests will
+-- not be run
+--
+validateLocalCmd
+    :: CW.ChainwebVersion
+    -> CW.ChainId
+      -- ^ Pact service chain id
+    -> Time Micros
+      -- ^ current integral time (micros)
+    -> Gas
+      -- ^ initial gas cost
+    -> TransactionEnv p
+      -- ^ the tx environment
+    -> Command (Payload PublicMeta ParsedCode)
+      -- ^ user command
+    -> NetworkId
+    -> TransactionM p (CommandResult [TxLog Value])
+validateLocalCmd v pscid now gas0 t cmd (NetworkId nid)
+    | chainIdToText pscid /= cid =
+      evalErr $ "invalid chain id: " <> cid
+    | not (timingsCheck bct cmd) =
+      evalErr "invalid ttl or creation time"
+    | (decimalPlaces gp) > 12 =
+      gasErr $ "gas price should be rounded to at most 12 places: " <> sshow gp
+    | fromText @CW.ChainwebVersion nid /= Just v =
```

Comment:
ChainId + Network Id vs. Chainweb Version + Sigcount not > 100  is checked in https://github.com/kadena-io/chainweb-node/blob/ba81dbd677ee9c40b8a5e7bda5d383534889fe70/src/Chainweb/Chainweb.hs#L535